cmake/FindLIBTBB.cmake

```diff
@@ -0,0 +1,44 @@
+# Copyright (C) 2024 Intel Corporation
+# Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+message(STATUS "Checking for module 'libtbb' using find_library()")
```

Comment:
Why do we need this file? Can't we just call `find_package(TBB components tbb)`?

Reply:
`find_package(TBB)` just calls the `FindLIBTBB.cmake` script

Reply:
Right, but I thought TBB already provides TBBconfig.cmake. Or do we use too old TBB version on our CI?

Reply:
Hmm, but to use `TBBconfig.cmake` we have to find the TBB package first, so to run `find_package(TBB)` we have to have it found already ... ;-)

Reply:
Fom what I understand find_package will look for the config files in the system just like find_library that you're using. The difference is only in case of non-standard location I think - you need to set `CMAKE_MODULE_PATH` instead of `CMAKE_PREFIX_PATH`.

Reply:
BTW, vcpkg docs recommend seeting CMAKE_TOOLCHAIN_FILE: https://github.com/microsoft/vcpkg?tab=readme-ov-file#using-vcpkg-with-cmake

Reply:
When you install tbb using vcpkg it says:
```
tbb provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(TBB CONFIG REQUIRED)
  target_link_libraries(main PRIVATE TBB::tbb TBB::tbbmalloc TBB::tbbmalloc_proxy)
```

I tested it locally and it works fine without findLIBTBB.cmake You just need to pass this CMAKE_TOOLCHAIN_FILE flag to cmake and change `find_package(LIBTBB REQUIRED tbb)` to `find_package(TBB CONFIG REQUIRED)` . Also, to get include dirs for TBB here: https://github.com/oneapi-src/unified-memory-framework/blob/c5785409841028022492f2c153b7689a7e920707/src/pool/CMakeLists.txt#L68 you need to do `get_target_property(LIBTBB_INCLUDE_DIRS TBB::tbbmalloc INTERFACE_INCLUDE_DIRECTORIES)`

Reply:
Same is true for findLIBHWLOC.cmake I think, we can probably remove that too.

Reply:
OK, I will try.

Reply:
@igchor It does not work:
```
CMake Error at CMakeLists.txt:134 (find_package):
  Found package configuration file:

    C:/Users/ldorau/work/vcpkg/packages/tbb_x64-windows/share/tbb/TBBConfig.cmake

  but it set TBB_FOUND to FALSE so package "TBB" is considered to be NOT
  FOUND.
```